src/main/java/uk/gov/hmcts/darts/arm/service/impl/TriggerArmRpoSearchServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+    private final ArmApiService armApiService;
+    private final UserIdentity userIdentity;
+
+    /**
+     * This method integrates various ARM RPO API calls to ultimately trigger a search. The results of that search are then processed by another automated
+     * task.
+     * <p/>
+     * If any of the underlying ArmRpoApi methods fail, ArmRpoApi will explicitly set the arm_rpo_execution_detail status to FAILED for that
+     * particular stage. So if you're thinking about adding any transactionality at the level of TriggerArmRpoSearchServiceImpl please be careful to avoid
+     * unwanted arm_rpo_execution_detail rollbacks.
+     */
+    @Override
+    public void triggerArmRpoSearch() {
+        log.debug("Triggering ARM RPO search flow...");
+
+        var userAccountEntity = userIdentity.getUserAccount();
+
+        final Integer executionId = armRpoService.createArmRpoExecutionDetailEntity(userAccountEntity)
+            .getId();
+
+        // armBearerToken may be null, but we'll let the lower level service methods deal with that by handling the resultant HTTP exception
+        final String armBearerToken = armApiService.getArmBearerToken();
+
+        armRpoApi.getRecordManagementMatter(armBearerToken,
+                                            executionId,
+                                            userAccountEntity);
+
+        // We expect getRecordManagementMatter() to populate the matter id as a side effect, so refresh the entity to get the updated value
+        final String matterId = armRpoService.getArmRpoExecutionDetailEntity(executionId)
+            .getMatterId();
+        armRpoApi.getIndexesByMatterId(armBearerToken,
+                                       executionId,
+                                       matterId,
+                                       userAccountEntity);
+
+        armRpoApi.getStorageAccounts(armBearerToken,
+                                     executionId,
+                                     userAccountEntity);
+
+        armRpoApi.getProfileEntitlements(armBearerToken,
+                                         executionId,
+                                         userAccountEntity);
+
+        armRpoApi.getMasterIndexFieldByRecordClassSchema(armBearerToken,
+                                                         executionId,
+                                                         ArmRpoHelper.getMasterIndexFieldByRecordClassSchemaPrimaryRpoState(),
+                                                         userAccountEntity);
+
+        String searchName = armRpoApi.addAsyncSearch(armBearerToken,
+                                                     executionId,
+                                                     userAccountEntity);
+
+        armRpoApi.saveBackgroundSearch(armBearerToken,
+                                       executionId,
+                                       searchName,
+                                       userAccountEntity);
```

Comment:
As these do not seem to depend on each other could they be made to run in parallel?

Reply:
Each of these methods set various fields on the underlying `ArmRpoExecutionDetailEntity` record, including the `armRpoState()` and `armRpoStatus` fields. So at this time, sequential calls are required by design (https://tools.hmcts.net/confluence/display/DMP/ARM+RPO).

Also I understand this code will only be run once a day by its corresponding automated task, so in this case, I'm not sure the perf improvements are worth the extra complexity of making the calls async.

Reply:
Following the above I'm happy with this approach :)